carp.common/src/commonMain/kotlin/dk/cachet/carp/common/application/tasks/WebTask.kt

```diff
@@ -19,45 +19,20 @@ data class WebTask(
     override val name: String,
     /**
      * The URL of the web page which contains the task to be performed.
-     * The URL may contain [UrlVariable] patterns which will be replaced with the corresponding values by the client runtime.
+     * The URL may contain [URL.Variable] patterns which will be replaced with the corresponding values by the client runtime.
      */
     val url: String,
     override val description: String? = null,
     override val measures: List<Measure> = emptyList(),
 ) : TaskConfiguration<NoData> // The execution of the task is delegated to a web page, so this task uploads no data.
 {
-    companion object
-    {
-        private fun markup( name: String ) = name.uppercase().replace( ' ', '-' ).let { "{{$it}}" }
-    }
-
-
-    enum class UrlVariable( val pattern: String )
-    {
-        /**
-         * Uniquely identifies the participant in the study.
-         */
-        PARTICIPANT_ID( markup( "participant id" ) ),
-
-        /**
-         * Uniquely identifies the deployment (group of participants and devices) of the study.
-         */
-        DEPLOYMENT_ID( markup( "deployment id" ) ),
-
-        /**
-         * Identifies the condition, defined by the study protocol, which caused the [WebTask] to be triggered.
-         */
-        TRIGGER_ID( markup( "trigger id" ) )
-    }
-
-
     /**
      * Replace the variables in [url] with the specified runtime values, if the variables are present.
+     *
+     * Propagates construction of [url] to [URL]. Kept for backwards compatibility.
```

Comment:
Don't _document_ implementation details. If relevant (I don't think it is here) use _comments_ for that. Consider documentation part of the API, it's supposed to _hide_ (encapsulate) implementation details.